core/trino-main/src/main/java/io/trino/tracing/TracingMetadata.java

```diff
@@ -0,0 +1,1393 @@
+    {
+        Span span = startSpan("getMaxWriterTasks", catalogName);
+        try (var ignored = scopedSpan(span)) {
+            return delegate.getMaxWriterTasks(session, catalogName);
+        }
+    }
+
+    private Span startSpan(String methodName)
+    {
+        return tracer.spanBuilder("Metadata." + methodName).startSpan();
+    }
+
+    private Span startSpan(String methodName, String catalogName)
+    {
+        return startSpan(methodName)
+                .setAttribute(TrinoAttributes.CATALOG, catalogName);
+    }
+
+    private Span getStartSpan(String methodName, Optional<String> catalog)
+    {
+        return startSpan(methodName)
+                .setAllAttributes(attribute(TrinoAttributes.CATALOG, catalog));
+    }
+
+    private Span startSpan(String methodName, CatalogSchemaName schema)
+    {
+        return startSpan(methodName)
+                .setAttribute(TrinoAttributes.CATALOG, schema.getCatalogName())
+                .setAttribute(TrinoAttributes.SCHEMA, schema.getSchemaName());
+    }
+
+    private Span startSpan(String methodName, QualifiedObjectName table)
+    {
+        return startSpan(methodName)
+                .setAttribute(TrinoAttributes.CATALOG, table.getCatalogName())
+                .setAttribute(TrinoAttributes.SCHEMA, table.getSchemaName())
+                .setAttribute(TrinoAttributes.TABLE, table.getObjectName());
+    }
+
+    private Span startSpan(String methodName, QualifiedTablePrefix prefix)
+    {
+        return startSpan(methodName)
+                .setAttribute(TrinoAttributes.CATALOG, prefix.getCatalogName())
+                .setAllAttributes(attribute(TrinoAttributes.SCHEMA, prefix.getSchemaName()))
+                .setAllAttributes(attribute(TrinoAttributes.TABLE, prefix.getTableName()));
+    }
+
+    private Span startSpan(String methodName, String catalogName, ConnectorTableMetadata tableMetadata)
+    {
+        return startSpan(methodName)
+                .setAttribute(TrinoAttributes.CATALOG, catalogName)
+                .setAttribute(TrinoAttributes.SCHEMA, tableMetadata.getTable().getSchemaName())
+                .setAttribute(TrinoAttributes.TABLE, tableMetadata.getTable().getTableName());
+    }
+
+    private Span startSpan(String methodName, TableHandle handle)
+    {
+        Span span = startSpan(methodName);
+        if (span.isRecording()) {
+            span.setAttribute(TrinoAttributes.CATALOG, handle.getCatalogHandle().getCatalogName());
+            span.setAttribute(TrinoAttributes.HANDLE, handle.getConnectorHandle().toString());
+        }
+        return span;
+    }
+
+    private Span startSpan(String methodName, TableExecuteHandle handle)
+    {
+        Span span = startSpan(methodName);
+        if (span.isRecording()) {
+            span.setAttribute(TrinoAttributes.CATALOG, handle.getCatalogHandle().getCatalogName());
+            span.setAttribute(TrinoAttributes.HANDLE, handle.getConnectorHandle().toString());
```

Comment:
is the handle going to be too long in text form or possibly leak push down sql?

Reply:
We print these in plans, so it's probably fine